hal/sparcv8leon3/gaisler/timer.c

```diff
@@ -0,0 +1,127 @@
+/*
+ * Phoenix-RTOS
+ *
+ * Operating system loader
+ *
+ * Timer controller
+ *
+ * Copyright 2022 Phoenix Systems
+ * Author: Lukasz Leczkowski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <hal/hal.h>
+
+/* Timer control bitfields */
+
+#define TIMER_ENABLE      (1 << 0)
+#define TIMER_PERIODIC    (1 << 1)
+#define TIMER_LOAD        (1 << 2)
+#define TIMER_INT_ENABLE  (1 << 3)
+#define TIMER_INT_PENDING (1 << 4)
+#define TIMER_CHAIN       (1 << 5)
+
+#define TIMER_DEFAULT_FREQ 1000
+
+
+/* clang-format off */
+
+enum { timer1 = 0, timer2, timer3, timer4 };
+
+/* clang-format on */
+
+
+static struct {
+	vu32 *gptimer0_base;
+	vu32 *gptimer1_base;
+	volatile time_t time;
+	u32 ticksPerFreq;
+} timer_common;
+
+
+static int timer_isr(unsigned int irq, void *data)
+{
+	vu32 st = *(timer_common.gptimer0_base + GPT_TCTRL1) & TIMER_INT_PENDING;
+
+	if (st != 0) {
+		++timer_common.time;
+		/* Clear irq status */
+		*(timer_common.gptimer0_base + GPT_TCTRL1) |= TIMER_INT_PENDING;
+		*(timer_common.gptimer0_base + GPT_TCTRL1) &= ~TIMER_INT_PENDING;
```

Comment:
Perhaps a brief comment here why we set and clear this bit? This is not obvious ;)